tests/Unit/Evolution/DgSubcell/Test_SliceData.cpp

```diff
@@ -0,0 +1,157 @@
+#include <cstddef>
+#include <numeric>
+#include <vector>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/SliceIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/DirectionMap.hpp"
+#include "Evolution/DgSubcell/SliceData.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::subcell {
+namespace {
+namespace Tags {
+struct Scalar : db::SimpleTag {
+  using type = ::Scalar<DataVector>;
+};
+
+template <size_t Dim>
+struct Vector : db::SimpleTag {
+  using type = tnsr::I<DataVector, Dim>;
+};
+}  // namespace Tags
+
+template <size_t Dim>
+void check_slice(
+    const Index<Dim>& volume_extents, const Index<Dim>& slice_extents,
+    const Direction<Dim>& direction, const size_t fixed_index,
+    const size_t ghost_slice, const size_t component_index,
+    const Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>>& volume_vars,
+    const DirectionMap<Dim, std::vector<double>>& sliced_data) noexcept {
+  const size_t volume_offset = component_index * volume_extents.product();
+  const size_t slice_offset = component_index * slice_extents.product();
+  CAPTURE(volume_offset);
+  CAPTURE(slice_offset);
+
+  for (SliceIterator
+           volume_it(volume_extents, direction.dimension(), fixed_index),
+       slice_it(slice_extents, direction.dimension(), ghost_slice);
+       volume_it; ++volume_it, ++slice_it) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    CHECK(*(volume_vars.data() + volume_offset + volume_it.volume_offset()) ==
+          sliced_data.at(direction)[slice_offset + slice_it.volume_offset()]);
+  }
+}
+
+template <>
+void check_slice<1>(
+    const Index<1>& volume_extents, const Index<1>& slice_extents,
+    const Direction<1>& direction, const size_t fixed_index,
+    const size_t ghost_slice, const size_t component_index,
+    const Variables<tmpl::list<Tags::Scalar, Tags::Vector<1>>>& volume_vars,
+    const DirectionMap<1, std::vector<double>>& sliced_data) noexcept {
+  const size_t volume_offset = component_index * volume_extents.product();
+  const size_t slice_offset = component_index * slice_extents.product();
+  CAPTURE(volume_offset);
+  CAPTURE(slice_offset);
+  for (SliceIterator volume_it(volume_extents, direction.dimension(),
+                               fixed_index);
+       volume_it; ++volume_it) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    CHECK(*(volume_vars.data() + volume_offset + volume_it.volume_offset()) ==
+          sliced_data.at(direction)[slice_offset + ghost_slice]);
+  }
+}
+
+template <size_t Dim>
+void test_slice_data(const std::optional<size_t> do_not_slice_in_direction) {
+  CAPTURE(do_not_slice_in_direction.value_or(-1));
+  for (size_t number_of_ghost_points = 1; number_of_ghost_points < 3;
```

Comment:
Why only test for 1, 2 ghost points? Should the `slice_data` function have an ASSERT warning that larger values are unsupported?

Reply:
To keep the test runtime down

Reply:
I feel the test may be better (loosely defined, but I think better targeted to what is being done in this function) with more ghost points (at least 1,2,3), but fewer different mesh sizes... what do you think?

If you keep the code as-is, then I suggest adding a comment "// Test with 1, 2 ghost points only to reduce test runtime" or similar.